design/ErrorHandling.md

```diff
@@ -23,6 +23,141 @@ There are several types of WebGPU calls that get their errors handled differentl
 Implementations should provide a way to enable synchronous validation, for example via a debug shim or via the developer tools.
 The extra overhead needs to be low enough that applications can still run while being debugged.
 
+## *Fatal Errors*: Lost/Recovered Events
+
+<!-- calling this revision 5 -->
+
+```webidl
+partial interface GPUAdapter {
+    Promise<GPUDevice> requestDevice(GPUDeviceDescriptor descriptor,
+                                     GPUDeviceLostCallback onDeviceLost);
+};
+
+callback GPUDeviceLostCallback = void (GPUDeviceLostInfo info);
+
+interface GPUDeviceLostInfo {
+    readonly attribute GPUDevice device;
+    readonly attribute DOMString reason;
+};
+```
+
+`GPUAdapter.requestDevice` requests a device from the adapter.
+It returns a Promise which resolves when a device is ready.
+The Promise may not resolve for a long time - it resolves when the browser is ready for the application to bring up (or restore) its content.
+If the adapter is lost before the Promise resolves, the Promise rejects.
```

Comment:
Wouldn't this remove all possibilities to get data about the device before creating it, for example to know which extensions / limits are present and can be requested?

Reply:
Yes, @Kangz , it would remove the possibility of getting data about the device before creating it.  The same is true of the WebXR, other media based APIs and even getContext.  For those, you pass information about what the context/device you would like and the browser returns to you a context/device that meets the criteria.

Reply:
To be discussed after this PR.